server/aesdsocket.c

```diff
@@ -195,57 +207,91 @@ void *threadFunc(void* threadArg){
+          syslog(LOG_ERR, "File write system call failed \r\n");
+          returnStatus = pthread_mutex_unlock(&mutexLock);                                           // Unlock Mutex if the write operation failed //
+          exitThread(param, recBuffer);
+          return NULL;
+        }
       }
-      bytesPerPacket = write(fileFd_W_R, recBuffer, bytesChanged);                                 // Write the data to file Descriptor //
-      if(bytesPerPacket == STATUS_FAILURE){
-        syslog(LOG_ERR, "File write system call failed \r\n");
-        returnStatus = pthread_mutex_unlock(&mutexLock);                                           // Unlock Mutex if the write operation failed //
+      bytesToBeWritten += bytesPerPacket;
+      noOfBytesinBuffer -= (ind + 1);
+      bytesChanged = bytesToBeWritten;
+      trsBuffer = (char*)malloc(bytesChanged);                                                   // Allocate memory to store the read data //
+      lseek(fileFd_W_R, 0, SEEK_SET);                                                            // Set cursor to the begining of the descriptor //
+      int bytesToSend = read(fileFd_W_R, trsBuffer, bytesChanged);                               // Read the data back again from the descriptor //
+      if(bytesToSend == STATUS_FAILURE){
+        syslog(LOG_ERR, "Cannot read from file \r\n");
+        returnStatus = pthread_mutex_unlock(&mutexLock);                                         // Unlock mutex if the read operation failed //
         exitThread(param, recBuffer);
         return NULL;
       }
       else{
-        bytesToBeWritten += bytesPerPacket;
-        noOfBytesinBuffer -= (ind + 1);
-        bytesChanged = bytesToBeWritten;
-        trsBuffer = (char*)malloc(bytesChanged);                                                   // Allocate memory to store the read data //
-        lseek(fileFd_W_R, 0, SEEK_SET);                                                            // Set cursor to the begining of the descriptor //
-        int bytesToSend = read(fileFd_W_R, trsBuffer, bytesChanged);                               // Read the data back again from the descriptor //
-        if(bytesToSend == STATUS_FAILURE){
-          syslog(LOG_ERR, "Cannot read from file \r\n");
-          returnStatus = pthread_mutex_unlock(&mutexLock);                                         // Unlock mutex if the read operation failed //
+        ssize_t bytesSent = send(clientFd, trsBuffer, bytesToSend, 0);                           // Send data over socket if the read is successful //
+        if(bytesSent == STATUS_FAILURE){
+          syslog(LOG_ERR, "Send failed \r\n");
+          returnStatus = pthread_mutex_unlock(&mutexLock);                                       // Unlock mutex if the send operation failed //
           exitThread(param, recBuffer);
           return NULL;
         }
+        else if (bytesSent == bytesToSend){
+          syslog(LOG_INFO, "Data sent properly \r\n");
```

Comment:
@Vignesh-Vadivel I'd add some logging here about what you sent to help troubleshoot the failure of your sockettest run